cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-verify-cancel.cy.js

```diff
@@ -0,0 +1,137 @@
+        BulkEditActions.verifyRowIcons();
+        BulkEditActions.verifyBulkEditForm();
+
+        BulkEditActions.selectOption('Patron group');
+        // Click "Select Option" dropdown => Select "Patron group" option
+        BulkEditActions.fillPatronGroup('staff (Staff Member)');
+
+        // Сlick "Confirm changes" option
+        BulkEditActions.confirmChanges();
+        BulkEditActions.verifyAreYouSureForm(1, user.username);
+        BulkEditActions.clickX();
+        BulkEditActions.verifyBulkEditForm();
+
+        BulkEditActions.closeBulkEditInAppForm();
+        BulkEditSearchPane.verifyErrorLabel(usernamesFileName, 1, 1);
+        BulkEditSearchPane.verifyPaneRecordsCount(1);
+        BulkEditSearchPane.verifyNonMatchedResults();
+        BulkEditSearchPane.verifyMatchedResults(user.username);
+
+        BulkEditActions.openActions();
+        BulkEditActions.downloadMatchedRecordsExists();
+        BulkEditActions.downloadErrorsExists();
+        BulkEditActions.openInAppStartBulkEditFrom();
+        BulkEditActions.verifyBulkEditForm();
+
+        BulkEditActions.selectOption('Email');
+        BulkEditActions.replaceWithIsDisabled();
+        const oldEmailDomain = 'test@example.com';
+        const newEmailDomain = 'test@example.org)';
+        BulkEditActions.replaceEmail(oldEmailDomain, newEmailDomain);
+        BulkEditSearchPane.isConfirmButtonDisabled(false);
+
+        BulkEditActions.confirmChanges();
+        // Click "Keep editing" button on  the "Are you sure?" form
+        BulkEditActions.clickKeepEditingBtn();
+        BulkEditActions.verifyBulkEditForm();
+        BulkEditActions.closeBulkEditInAppForm();
+        BulkEditSearchPane.verifyMatchedResults(user.username);
+        BulkEditSearchPane.verifyErrorLabel(usernamesFileName, 1, 1);
+
+        BulkEditSearchPane.openLogsSearch();
+        BulkEditSearchPane.verifyLogsPane();
+        BulkEditSearchPane.openIdentifierSearch();
+
+        // Click "Actions" menu
+        BulkEditActions.openActions();
+        BulkEditActions.openInAppStartBulkEditFrom();
+
+        BulkEditActions.verifyRowIcons();
+        BulkEditActions.verifyBulkEditForm();
+        BulkEditActions.selectOption('Expiration date');
+
+        // Click on the date picker plugin => Select a date in the future differ from the current expiration date (for example, <current expiration date + 3 months>)
+        BulkEditActions.fillExpirationDate(nextThreeMonthsDate);
+        BulkEditActions.verifyPickedDate(nextThreeMonthsDate);
+
+        BulkEditActions.confirmChanges();
+        BulkEditActions.commitChanges();
+        BulkEditSearchPane.waitFileUploading();
+        BulkEditActions.verifySuccessBanner(1);
+        // Navigate to the "Users" app => Find and open updated user record=> Make sure that changes were applied
+        TopMenuNavigation.navigateToApp('Users');
+        UsersSearchPane.searchByUsername(user.username);
+        Users.verifyPatronGroupOnUserDetailsPane('staff');
```

Comment:
were patron groups the changes you applied, or should we verify for another change?